### PR TITLE
SDK 3.35.0 : Adds version bump and changelog

### PR DESCRIPTION
**Pull request by NaluTripician**
# Pull Request Template

## Description

This bumps the SDK version 3.35.0

Public contract change: 
image

Preview contract change: 
image

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -12,6 +12,35 @@ Preview features are treated as a separate branch and will not be included in th
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### <a name="3.34.0-preview"/> [3.35.0-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.35.0-preview) - 2023-06-19
+
+### Added
+- [3836](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3836) Integrated cache: Adds BypassIntegratedCache to DedicatedGatewayRequestOptions
+- [3909](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3909) Query: Refactors EnableOptimisticDirectExecution to true by default in Preview mode
+
+#### Fixed
```
Comment (sourabh1007) [inline changelog.md:21]: You can remove all 3 PR entries from the changelog. these are not related to SDK functionality. I should have marked them internal while creating these PRs

File: changelog.md [hunk 2]
```diff
@@ -12,6 +12,35 @@ Preview features are treated as a separate branch and will not be included in th
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### <a name="3.34.0-preview"/> [3.35.0-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.35.0-preview) - 2023-06-19
+
+### Added
+- [3836](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3836) Integrated cache: Adds BypassIntegratedCache to DedicatedGatewayRequestOptions
+- [3909](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3909) Query: Refactors EnableOptimisticDirectExecution to true by default in Preview mode
```
Comment (ealsur) [inline changelog.md:19]:
```suggestion
- [3909](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3909) Query: Adds EnableOptimisticDirectExecution in QueryRequestOptions enabled by default
```

File: changelog.md [hunk 3]
```diff
@@ -12,6 +12,35 @@ Preview features are treated as a separate branch and will not be included in th
...
+#### Fixed 
+- [3864](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3864) NugetPackage: Removes ThirdPartyNotice.txt from content and contentFiles folders
+- [3866](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3866) CosmosClient: Fixes missing Trace when converting HTTP Timeout to 503
+- [3879](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3879) Subpartitioning: Fixes handling of split physical partitions
+- [3907](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3907) Query: Fixes empty property name parsing exception
+
+#### Added
+- [3860](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3860) Documentation: Adds see also link to Container.CreateTransactionalBatch
+- [3852](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3852) Query: Adds type-markers with count and length for large arrays
+- [3838](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3838) Benchmarking: Adds use of ARM Templates for benchmarking
```
Comment (ealsur) [inline changelog.md:37]: Remove

File: changelog.md [hunk 4]
```diff
@@ -12,6 +12,35 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [3877](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3877) Direct Package Upgrade 3.31.0: Refactors code to make compatible with latest direct
```
Comment (ealsur) [inline changelog.md:38]: This PR actually does multiple things, so better to split it into different bullet points:

```
- [3877](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3877) Regions: Adds Malaysia South and Italy North
- [3877](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3877) Distributed Tracing: Enables Distributing Tracing for Operations
 ```

@sourabh1007 Is the above correct?

File: changelog.md [hunk 5]
```diff
@@ -12,6 +12,35 @@ Preview features are treated as a separate branch and will not be included in th
...
+
+### <a name="3.35.0"/> [3.35.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.35.0) - 2023-06-19
+
+#### Fixed 
+- [3864](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3864) NugetPackage: Removes ThirdPartyNotice.txt from content and contentFiles folders
+- [3866](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3866) CosmosClient: Fixes missing Trace when converting HTTP Timeout to 503
+- [3879](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3879) Subpartitioning: Fixes handling of split physical partitions
+- [3907](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3907) Query: Fixes empty property name parsing exception
+
+#### Added
```
Comment (ealsur) [inline changelog.md:29]: https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3918 was marked as Internal but actually has several public facing changes:

```
- [3918](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3918) Regions: Adds Israel Central
- [3918](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3918) CosmosClient: Fixes SynchronizationLockException when disposing client with requests in-flight.
```

**Review by kirankumarkolli — APPROVED** [review]

File: changelog.md (hunk 4 above)
Comment (sourabh1007) [reply changelog.md]: no, you can remove point related to distributed tracing..again it get enabled only in test. it is not related to actual feature.

Comment (sourabh1007) [reply changelog.md]: instead put this  

- [3887](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3887) Distributed Tracing: Setting DisplayName for an operation level activity as _`<operationname><space><containername>`_
